src/Command/Update/ExecuteCommand.php

```diff
@@ -151,42 +154,49 @@ protected function execute(InputInterface $input, OutputInterface $output)
                     )
                 );
             }
+        } else {
 
-            return 0;
-        }
-
-        $maintenanceMode = $this->state->get('system.maintenance_mode', false);
-
-        if (!$maintenanceMode) {
-            $this->getIo()->info($this->trans('commands.site.maintenance.description'));
-            $this->state->set('system.maintenance_mode', true);
-        }
-
-        try {
-            $this->runUpdates(
-                $updates
-            );
-
-            // Post Updates are only safe to run after all schemas have been updated.
-            if (!$this->getUpdates()) {
-                $this->runPostUpdates();
+            if (!$currentMaintenanceMode) {
+                $this->getIo()->info($this->trans('commands.site.maintenance.description'));
+                $this->state->set('system.maintenance_mode', true);
+                $currentMaintenanceMode = true;
             }
-        } catch (\Exception $e) {
-            watchdog_exception('update', $e);
-            $this->getIo()->error($e->getMessage());
-            return 1;
-        }
 
-        if (!$maintenanceMode) {
-            $this->state->set('system.maintenance_mode', false);
-            $this->getIo()->info($this->trans('commands.site.maintenance.messages.maintenance-off'));
+            try {
+                $this->runUpdates(
+                    $updates
+                );
+            } catch (\Exception $e) {
+                watchdog_exception('update', $e);
+                $this->getIo()->error($e->getMessage());
+                return 1;
+            }
         }
+        
 
+        // Post Updates are only safe to run after all schemas have been updated.
         if (!$this->getUpdates()) {
+            $postUpdates = $this->postUpdateRegistry->getPendingUpdateInformation();
```

Comment:
It looks like you set maintenance mode, but it will be reverted only when where are $postUpdates. In other case the website will stay in maintenance mode.
Actually, there is better mechanism for  maintenance mode in the Console core, it is merged, but it is not tagged yet. I asked to create a new tag.

Reply:
Check the way we will activate maintenance mode

```
protected function configure()
    {
        $this
            ->setName('update:entities')
            ->setDescription($this->trans('commands.update.entities.description'))
            ->enableMaintenance()
            ->setAliases(['upe']);
    }

```

Reply:
it looks much better with this new helper.
I had updated the code, do you want me to rebase the pr ?